src/site/content/en/notifications/use-push-notifications-to-engage-users/index.md

```diff
@@ -0,0 +1,161 @@
+
+### Part 1: Registration and subscription
+
+1.  A client app registers a service worker with `ServiceWorkerContainer.register()`. The registered service worker will continue to run in the background when the client is inactive. 
+
+    Client code: 
+
+    ```js
+    navigator.serviceWorker.register('sw.js');
+    ```
+
+1.  The client requests permission from the user to send notifications.
+
+    Client code: 
+
+    ```js
+    Notification.requestPermission();
+    ```
+
+1.  To enable push notifications, the service worker uses `PushManager.subscribe()` to create a push subscription. In the call to `PushManager.subscribe()`, the service worker supplies the app's API key as an identifier.
+
+    Client code:
+
+    ```js/0-1/0/
+    navigator.serviceWorker.register('sw.js').then(sw => {
+      sw.pushManager.subscribe({ /* API key */ });
+    });
+    ```
+
+    Push services like Firebase Cloud Messaging operate on a subscription model. When a service worker subscribes to a push service by calling `PushManager.subscribe()`, the push service creates a URL unique to that service worker. The URL is known as a subscription endpoint. 
+
+1.  The client sends the subscription endpoint to the app server. 
+
+    Client code:
+
+    ```js/1-3/1/
+    navigator.serviceWorker.register('sw.js').then(sw => {
+      sw.pushManager.subscribe({ /* API key */ }).then(subscription => {
+        sendToServer(subscription, '/new-subscription', 'POST');
+      });
+    });
+    ```
+
+    Server code: 
+
+    ```js
+    app.post('/new-subscription', (request, response) => {
+      // extract subscription from request
+      // send 'OK' response
+    });
+    ```
+
+Part 2: Send Notification
+
+1.  The web server sends a notification to the subscription endpoint.
+
+    Server code:
+
+    ```js
+    const webpush = require('web-push');
+
+    let options = { /* config info for cloud messaging and API key */ };  
+    let subscription = { /* subscription created in Part 1*/ };
+    let payload = { /* notification */ };
+
+    webpush.sendNotification(sub, payload, options);
+    ```
+
+1.  Notifications sent to the subscription endpoint fire push events with the service worker as the target. The service worker receives the message and displays it to the user as a notification.
+
+    Service worker code:
+
+    ```js
+    self.addEventListener('push', (event) => {
+      let title = { /* get notification title from event data */ }
+      let options = { /* get notification options from event data */ }
+      showNotification(title, options);
+    })
+    ```
+
+1.   The user interacts with the notification, making the web app active if it wasn't already.
+
+## Next steps
+
+As a next step, implement push notifications! 
+
+We've created a series of codelabs to guide you through each step of the process:
```

Comment:
Since this is right at the end of the article, I might rely on the `codelabs` YAML, which will generate a callout, rather than also including these asides.